src/uu/ls/src/ls.rs

```diff
@@ -1084,7 +1079,8 @@ impl Config {
 
         let dired = options.get_flag(options::DIRED);
         if dired && format != Format::Long {
-            return Err(Box::new(LsError::ConflictingArgumentDired));
+            // --dired implies --format=long
+            format = Format::Long;
         }
```

Comment:
Nope, it appears that GNU ls has some different semantics:
```console
$ ls --version
ls (GNU coreutils) 9.4
[…]
$ ls --dired                         
CODE_OF_CONDUCT.md  CONTRIBUTING.md  Cargo.toml      GNUmakefile  Makefile       README.md  b      build.rs  deny.toml  foo   fuzz      oranda.json    showargs  target  util
COMMANDS.txt        Cargo.lock       DEVELOPMENT.md  LICENSE      Makefile.toml  a          build  check.sh  docs       foo2  often.sh  renovate.json  src       tests
$ ls --dired --format=single-column
CODE_OF_CONDUCT.md
COMMANDS.txt
CONTRIBUTING.md
Cargo.lock
Cargo.toml
DEVELOPMENT.md
GNUmakefile
LICENSE
Makefile
Makefile.toml
README.md
a
b
build
build.rs
check.sh
deny.toml
docs
[… more lines …]
```

Perhaps it only sets the default value?

Reply:
well, i worked on the GNU implementation (with the maintainer) :)
https://github.com/coreutils/coreutils/commit/8367b95a13b96b96db93a7339c094f813af877c9
(as author, I can share this link)

Reply:
Ah! Then it's by definition the same behavior. Makes sense :D

Maybe `ls` should warn when `--format=single-column` is ignored due to this? But that's a topic out of scope for this PR.

Reply:
yeah, i will have a look in a different PR (and maybe add a test on the GNU side)